internal/cortex/frontend/transport/handler.go

```diff
@@ -153,22 +145,14 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	r.Body = http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize)
 	r.Body = io.NopCloser(io.TeeReader(r.Body, &buf))
 
-	// Check if caching is enabled.
-	if f.lruCache != nil {
-		// Store query expression.
-		queryExpressionNormalized = f.regex.ReplaceAllString(r.URL.Query().Get("query"), " ")
+	urlQuery = r.URL.Query()
 
-		// Store query time range length.
-		queryExpressionRangeLength = getQueryRangeSeconds(r)
-
-		// Check if query in cache and whether value exceeds time range length.
-		if value, ok := f.lruCache.Get(queryExpressionNormalized); ok && value.(int) >= queryExpressionRangeLength {
+	// Check if query is cached
+	if f.failedQueryCache != nil {
+		cached, message := f.failedQueryCache.CallQueryHitCache(urlQuery)
```

Comment:
`CallQueryHitCache` is a weird name. You can drop `Call` prefix. It's literally calling a function.